### ISSUE TITLE
v0.21.3 does not set charset in Content-Type header

### ISSUE THREAD

**Issue by ddolcimascolo** [axios/axios#4016]
#### Describe the bug

Axios 0.21.3 (0.21.2 too, actually) does not set `chartset=utf-8` in `Content-Type` header on POST requests, using an object as `data`

#### To Reproduce

I've created a Runkit: https://runkit.com/ddolcimascolo/axios-0.21.3-content-type-issue

```js
const http = require("http"),
      axios0_21_3 = require('axios@0.21.3'),
      axios0_21_1 = require('axios@0.21.1');

function doPOSTRequest(axios, version) {
  return axios({
    method: 'post',
    url: 'http://127.0.0.1:8080',
    data: { my: 'object' }
  })
    .then(res => console.log(`Content-Type with v${version}: ${res.data['content-type']}`));
}

const server = http
  .createServer((req, res) => {
    res.write(JSON.stringify(req.headers));
    res.end();
  })
  .listen(8080, async() => {
    await doPOSTRequest(axios0_21_1, '0.21.1');
    await doPOSTRequest(axios0_21_3, '0.21.3');
    
    server.close();
  });
```

This code will yield
```
"Content-Type with v0.21.1: application/json;charset=utf-8"
"Content-Type with v0.21.3: application/json"
```

#### Expected behavior

`Content-Type` header should contain `charset=utf-8`

#### Environment
 - Axios Version [0.21.3]
 - Adapter [HTTP]
 - Browser [N/A]
 - Browser Version [N/A]
 - Node.js Version [14]
 - OS: [Ubuntu 20.04]
 - Additional Library Versions [None]

#### Additional context/Screenshots

N/A

**Comment by ddolcimascolo** [comment]
OK, I just found https://github.com/axios/axios/issues/2154 that is actually mentioned in 0.21.2 changelog. I've missed it the first time because it's in "Internal/Tests" section.

IMHO it's not OK to "hide" this in the "Internal" section. It's not an internal change at all, it's a breaking change in Axios behavior. At least you should have mentioned it in a "Breaking changes" section, and even better you should not have released a patch version... I know you're still in 0.x so you can break stuff, etc. but Axios is a lib used by hundreds of projects and breaking things is not cool.

**Comment by jasonsaayman** [comment]
@ddolcimascolo Sorry about this, I know some things may have fallen through the cracks, this was a release that spanned almost a year so not everything was fresh in my mind. We did not intentionally hide this, in fact I did not think that this was a largely impactful change, I will try be more cognisant of this in the future.

**Comment by ddolcimascolo** [comment]
I know @jasonsaayman. No worries

Thx for maintaining Axios!

**Comment by gfortaine** [comment]
@ddolcimascolo It should be fixed by 0.21.4 https://github.com/axios/axios/pull/4020
